library/alloc/src/sync.rs

```diff
@@ -522,16 +536,67 @@ impl<T> Arc<T> {
     /// let x = Arc::new(3);
     /// let y = Arc::clone(&x);
     ///
+    /// // Two threads calling `unwrap_or_drop` on both clones of an `Arc`:
     /// let x_unwrap_thread = std::thread::spawn(|| Arc::unwrap_or_drop(x));
     /// let y_unwrap_thread = std::thread::spawn(|| Arc::unwrap_or_drop(y));
     ///
     /// let x_unwrapped_value = x_unwrap_thread.join().unwrap();
     /// let y_unwrapped_value = y_unwrap_thread.join().unwrap();
     ///
+    /// // One of the threads is guaranteed to receive the inner value:
     /// assert!(matches!(
     ///     (x_unwrapped_value, y_unwrapped_value),
     ///     (None, Some(3)) | (Some(3), None)
     /// ));
+    ///
+    ///
```

Comment:
If the three empty lines in the example are here to separate two different examples, you should make two, with maybe a sentence describing the examples before each of them.

Reply:
I wasn’t sure of the correct way to format this since I haven’t found any precedence of multiple example code blocks yet. Perhaps there are some examples like this somewhere in the standard library.

Reply:
Here are some examples: [`unsafe` keyword](https://doc.rust-lang.org/nightly/std/keyword.unsafe.html), [`std::vec::Vec`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html), [`std::vec::Vec::set_len()`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.set_len).

Reply:
Thanks, I split them up now ^^

Reply:
This is good now, thanks !